modules/host/host.go

```diff
@@ -148,6 +158,38 @@ func (h *Host) Capacity() int64 {
 	return h.spaceRemaining
 }
 
+// Close shuts down the host, preparing it for garbage collection.
+func (h *Host) Close() error {
+	// The order in which things are closed has been explicitly chosen to
+	// minimize turbulence in the event of an error.
+
+	// Save the latest host state.
+	err := func() error {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+		return h.save()
+	}()
```

Comment:
I think the latter approach (locking and unlocking with no branches in between) is a preferable style as long as there are a few number of lines (i.e. 1) between the lock and unlock. Otherwise it would be too easy to accidentally add a line that branches without unlocking.